packages/kit/src/ignore.ts

```diff
@@ -35,6 +28,31 @@ export function isIgnored (pathname: string): boolean {
   return !!(relativePath && nuxt._ignore.ignores(relativePath))
 }
 
+export function resolveIgnorePatterns (relativePath?: string): string[] {
+  const nuxt = tryUseNuxt()
+
+  // Happens with CLI reloads
+  if (!nuxt) {
+    return []
+  }
+
+  if (!nuxt._ignorePatterns) {
+    nuxt._ignorePatterns = nuxt.options.ignore.flatMap(s => resolveGroupSyntax(s))
+
+    const nuxtignoreFile = join(nuxt.options.rootDir, '.nuxtignore')
+    if (existsSync(nuxtignoreFile)) {
+      const contents = readFileSync(nuxtignoreFile, 'utf-8')
+      nuxt._ignorePatterns.push(...contents.trim().split('\n'))
+    }
+  }
+
+  if (relativePath) {
+    return nuxt._ignorePatterns.map(p => p.startsWith('*') || p.startsWith('!*') ? p : relative(relativePath, resolve(nuxt.options.rootDir, p)))
```

Comment:
I haven't test locally but since tests passing, i think rootDir (vs ./server) is working well with nitro ignore patterns 👍🏼

Reply:
what do you mean? are you suggesting a change?

Reply:
i mean should be safe as long as tests are happy!